### ISSUE TITLE
Losses not decreasing over time

### ISSUE THREAD

**Issue by ManuBN786** [ArcticHare105/S3Diff#24]
I'm training on custom dataset for a 1X & 2X (256-->512) .

I already have the LR, HR pairs. 

So I removed the degradation_proc() to generate LR images and is using my own dataloader.

The rest of the setting are the same 

But the training losses are not decreasing

When I measure the psnr, ssim, FID , lpips they dont show any improvement on the validation dataset.

![Image](https://github.com/user-attachments/assets/0aa21cc5-84ed-4957-8539-58554894eba5)

So how to make the losses go down and finally converge ?

**Comment by ArcticHare105** [comment]
In my synthetic dataset, the loss plateaus after thousands of iterations; however, the performance continues to gradually improve. If your dataset is small or lacks diversity, consider augmenting it or combining it with synthetic data. Additionally, I recommend fine-tuning the model using our pre-trained weights, as this can help the model more easily converge to a local minimum.

**Comment by ManuBN786** [comment]
Using the pretrained model makes no difference !!

![Image](https://github.com/user-attachments/assets/95d6fda7-b095-4043-ab0f-20410bd96d1d)

Also the model is producing some whitish background features not there in the original ground truth!!

I'm training on BioSR  dataset by the way...

![Image](https://github.com/user-attachments/assets/07973846-5f1c-4c2e-aa82-43507f63a47b)

This is the LR train image


![Image](https://github.com/user-attachments/assets/74937984-407e-4d0a-8ca7-96f856e4f63f)

The GT HR image


![Image](https://github.com/user-attachments/assets/d80ed626-091c-470b-a90d-6517abcb0948)

The predicted SR image by the model

How to get rid of the whitish features ?

If I remove the GAN losses & train just with L2 and LPIPS loss then I can get rid of those whitish features in the background?

**Comment by ManuBN786** [comment]
I removed the GAN losses & finetuned using only with L2+LPIPS losses .

Despite that the performance is very poor as losses dont seem to decrease & converge.

Something wrong with the network itself !!

By the way, I have plenty of images in the dataset 101,000 pairs of (LR-HR) for training & 10,680 (LR-HR) pairs for testing.

![Image](https://github.com/user-attachments/assets/15c67c96-3094-4d14-9c8e-6d87af3a0040)

**Comment by ArcticHare105** [comment]
The GAN loss is a critical component of our model, and removing it is unlikely to yield promising results. Additionally, your training data appears to contain sparse useful foreground information, with a significant amount of background with noise. To address this, it is important to strike a balance between the foreground and background, ensuring that the model does not overly focus on the relatively simple background content.

**Comment by ManuBN786** [comment]
Thanks for the very quick reply !!

I have already trained it with the GAN loses too ! 

But it simply wont converge even after 50,000 iterations.

My LR-HR pairs look like this:

![Image](https://github.com/user-attachments/assets/485c3281-bc4c-4516-8f29-b99870a29bda)   ![Image](https://github.com/user-attachments/assets/e15e94d5-1445-4b8a-bdad-13ad916c1b68)

How do I train to get good SR results ?

**Comment by ArcticHare105** [comment]
You need to strike a balance between the foreground and background by introducing some losses or manually defining foreground/background regions, ensuring that the model does not overly focus on the relatively simple background content.

**Comment by ManuBN786** [comment]
No matter how many times I tried it didnt work.

So I'm closing this issue as I found other architectures and made it work
